tests/test_marquez_dag.py

```diff
@@ -10,42 +10,87 @@
-    log.debug("test_dummy_dag()")
-    execute_dag(dagbag.dags['test_dummy_dag'], '2020-01-08T01:00:00Z')
-    assert_marquez_calls(
-        client_mock(), namespace='test-marquez', owner='default_owner',
-        jobs=[(
-            'test_dummy_dag.test_dummy',
-            'BATCH',
-            'https://github.com/MarquezProject/marquez-airflow/blob/abcd1234/'
-            'tests/test_dags/test_dummy_dag.py', [], [],
-            {'sql': False},
-            'Test dummy DAG')],
-        job_runs=[('test_dummy_dag.test_dummy',
-                   '2020-01-08T01:00:00Z', '2020-01-08T01:02:00Z',
-                   {"external_trigger": False})],
-        run_ids=['00000000-0000-0000-0000-000000000000'])
-
-
-if __name__ == "__main__":
-    pytest.main([sys.argv[0]])
+import mock
+
+from airflow.models import (TaskInstance, DagRun)
+from airflow.operators.dummy_operator import DummyOperator
+from airflow.utils.db import provide_session
+from airflow.utils.dates import days_ago
+from airflow.utils import timezone
+from airflow.utils.state import State
+
+from marquez_airflow import DAG
+
+DEFAULT_DATE = timezone.datetime(2016, 1, 1)
+
+DEFAULT_ARGS = {
+    'owner': 'test_owner',
+    'depends_on_past': False,
+    'start_date': days_ago(1),
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'email': ['owner@test.com']
+}
+
+DAG = DAG(
+    'test_dag',
+    schedule_interval='@daily',
+    default_args=DEFAULT_ARGS,
+    description='''
+    A simple DAG to test the metadata extraction flow via Marquez.
+    '''
+)
+
+
+@pytest.fixture
+@provide_session
+def clear_db_airflow_dags(session=None):
+    session.query(DagRun).delete()
+    session.query(TaskInstance).delete()
+
+
+@mock.patch('marquez_airflow.DAG.get_or_create_marquez_client')
+@provide_session
+def test_marquez_dag(mock_get_or_create_marquez_client,
+                     clear_db_airflow_dags, session=None):
+    # (1) Mock the marquez client method calls
+    mock_marquez_client = mock.Mock()
+    mock_get_or_create_marquez_client.return_value = mock_marquez_client
+
+    # (2) Add task that will be marked as completed
+    completed = DummyOperator(
+        task_id='test_task_completed',
+        dag=DAG
+    )
+
+    # (3) Add task that will be marked as failed
+    failed = DummyOperator(
+        task_id='test_task_failed',
+        dag=DAG
+    )
+
+    # (4) Create DAG run and mark as running
+    dagrun = DAG.create_dagrun(
+        run_id="test_run_id_for_task_completed_and_failed",
+        execution_date=DEFAULT_DATE,
+        state=State.RUNNING)
+
+    assert mock_marquez_client.create_namespace.call_count == 1
+    assert mock_marquez_client.create_datasource.call_count == 0
+    assert mock_marquez_client.create_dataset.call_count == 0
+    assert mock_marquez_client.create_job.call_count == 2
+    assert mock_marquez_client.create_job_run.call_count == 2
+    assert mock_marquez_client.mark_job_run_as_started.call_count == 2
```

Comment:
We should set expectations to the mock calls about the parameters passed to those calls.

Reply:
We should make sure that those methods were called with the correct parameters

Reply:
Sure, let's be more concise with what we expect 👍